packages/amplify-codegen/src/commands/types.js

```diff
@@ -96,22 +95,30 @@ async function generateTypes(context, forceDownloadSchema, withoutInit = false,
         const introspection = path.extname(schemaPath) === '.json';
 
         try {
-          const output = await generateTypesHelper({
-            schema,
-            queries,
-            target,
-            introspection,
-          });
-          const outputs = Object.entries(output);
-
-          const outputPath = path.join(projectPath, generatedFileName);
-          if (outputs.length === 1) {
-            const [[, contents]] = outputs;
-            fs.outputFileSync(path.resolve(outputPath), contents);
+          if (target === 'swift' && fs.existsSync(outputPath) && fs.statSync(outputPath).isDirectory()) {
```

Comment:
This is the previous behavior for setting multiple files. https://github.com/aws-amplify/amplify-codegen/blob/9f11cfe396bacc603df84233a1b75feabcadaae6/packages/graphql-types-generator/src/generate.ts#L47C35-L47C102